tests/utils/tracked_container.py

```diff
@@ -92,5 +92,9 @@ def _lines_starting_with(logs: str, pattern: str) -> list[str]:
 
     def remove(self) -> None:
         """Kills and removes the tracked docker container."""
-        if self.container:
+        if self.container is None:
+            LOGGER.info("No container to remove")
+        else:
+            LOGGER.info(f"Removing container {self.container.name} ...")
             self.container.remove(force=True)
+            LOGGER.info(f"Container {self.container.name} removed")
```

Comment:
Just better logs.

There is some sort of code duplication between DockerRunner and TrackedContainer, but for now let's keep `tagging/` and `tests` dir completely independent of each other.